### PR TITLE
build UserSearchBar component functionality

### PR DESCRIPTION
**Pull request by PCapid3v**
https://trello.com/c/p5HWwTzJ/48-create-searchuserinput-component

### CODE REVIEW

**Review by PeachyOmnivore — COMMENTED** [review]
LGTM

File: src/components/userSearchBar/UserSearchBar.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      <Card>
+        <form>
+          <TextInput
+            icon={<SearchIcon />}
+            value={searchVal}
+            name="Search"
+            onChange={onChange}
+          />
+        </form>
+        {/* this card is used to display the results */}
```
Comment (Auenc) [inline src/components/userSearchBar/UserSearchBar.js:39]: let's remove this comment

**Review by Auenc — CHANGES_REQUESTED** [review]

File: src/service/apiClient.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ async function postPost(newPost) {
 }
 
 async function getUserByName(firstName) {
-  const res = await get(`users?first_name=${firstName}`);
+  const res = await get(`users?name=${firstName}`);
```
Comment (Uprising5034) [inline src/service/apiClient.js:32]: change `firstName` to `name`

**Review by Uprising5034 — DISMISSED** [review]

File: src/components/userSearchBar/UserSearchBar.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    setSearchVal(e.target.value);
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      try {
+        const res = await getUserByName(searchVal);
+        setSearchResults(res);
+      } catch (error) {
+        console.error("Error fetching user:", error);
```
Comment (Auenc) [inline src/components/userSearchBar/UserSearchBar.js:22]: I think we should display this error to the user. If something goes wrong, the user won't know why the search hasn't worked.

**Review by Auenc — CHANGES_REQUESTED** [review]

**Review by PCapid3v — COMMENTED** [review]
update all changes

File: src/components/userSearchBar/UserSearchBar.js (hunk 3 above)
Comment (PCapid3v) [reply src/components/userSearchBar/UserSearchBar.js:22]: That error should be handled by whoever is in charge of "display results"?or am I wrong? I don't want to step into someone else work...

Comment (Auenc) [reply src/components/userSearchBar/UserSearchBar.js:22]: I think it should be displayed in your Card, just like the search results are

**Review by Auenc — APPROVED** [review]

**Pull request merged**